### PR TITLE
Trialization

### PR DESCRIPTION
**Pull request by CBroz1**
Depends on:
- [interface PR](https://github.com/datajoint/element-interface/pull/32)
- [element-miniscope PR suggestion](https://github.com/datajoint/element-miniscope/pull/13#discussion_r860991623)

Added
- Trial data files
- ingestion functions first pass - needs testing bc element-interface version did not include `allow_direct_insert` arg
- `analysis.py` script for downstream analysis - adjusted for attribute differences
- fully functional notebooks **except 04**

Needs
- integration tests - Nice to have
- adjustment of scan time in `analysis.py`
- verification of fake equipment data in `notebooks/03`

Note: I have the following in my `.bash_profile` to expedite jupytext process, which motivated the `scripts/` -> `py_scripts/` rename. We could adopt another convention.
  ```bash
  alias jupysync='jupytext --sync notebooks/0*ipynb; mv notebooks/*py notebooks/py_scripts'
  ```

### CODE REVIEW

**Comment by CBroz1** [comment]
Drafting notebooks 05 and 07 now

**Comment by CBroz1** [comment]
@kabilar - ran most of both notebook 05 and 07, but ran into errors in each case where miniscope doesn't match structure from calcium imaging. If you use the `u24_` prefix on `dss-db`, you can pick up where I left off.

05 error:
```python
average_image = (miniscope.MotionCorrection.Summary & curation_key
                ).fetch1('average_image')
plt.imshow(average_image);
# ---
TypeError: Invalid shape (1, 600, 600) for image data
```

07 error
```python
analysis.ActivityAlignment.populate(display_progress=True)
# ---
UnknownAttributeError: Unknown column 'mask' in 'order clause'
```

File: workflow_miniscope/analysis.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+    def make(self, key):
+        # DEVNOTE : caimg->miniscope, no scan_datetime. so removed scan_time from fetch
+        #           safe to assume no sess-scan diff?
+        sess_time, nframes, frame_rate = (miniscope.RecordingInfo
+                                                     * session.Session
+                                                     & key
+                                                     ).fetch1('session_datetime',
+                                                              # 'scan_datetime',
+                                                              'nframes', 'fps')
```
Comment (ttngu207) [inline workflow_miniscope/analysis.py]:
```suggestion
        # DEVNOTE : caimg->miniscope, no scan_datetime. so removed scan_time from fetch
        #           safe to assume no sess-scan diff?
        sess_time, rec_time, nframes, frame_rate = (miniscope.RecordingInfo
                                                     * session.Session
                                                     & key
                                                     ).fetch1('session_datetime',
                                                              'recording_datetime',
                                                              'nframes', 'fps')
```

File: workflow_miniscope/analysis.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        scan_time = None # BEING ADDED TO ELEMENT
```
Comment (ttngu207) [inline workflow_miniscope/analysis.py]:
```suggestion
```

File: workflow_miniscope/analysis.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+        # Estimation of frame timestamps with respect to the session-start
+        # (to be replaced by timestamps retrieved from some synchronization routine)
+        scan_start = (scan_time - sess_time).total_seconds() if scan_time else 0
+        frame_timestamps = np.arange(nframes) / frame_rate + scan_start
+
```
Comment (ttngu207) [inline workflow_miniscope/analysis.py]:
```suggestion
        rec_start = (rec_time - sess_time).total_seconds() if rec_time else 0
        frame_timestamps = np.arange(nframes) / frame_rate + rec_start

```

**Review by ttngu207 — CHANGES_REQUESTED** [review]

**Comment by kabilar** [comment]
Hi @CBroz1, just a heads up, I made the mistake of directly committing Thinh's suggestions to your fork/branch.

**Comment by kabilar** [comment]
Thank you for these updates, @CBroz1!

**Comment by kabilar** [comment]
I removed trial 50 since the start time occurred before trials 47-49.

**Pull request merged**